## Question 1: [Index] S&P 500 Stocks Added to the Index

### Which year had the highest number of additions (starting from 2020)?

In [73]:
import requests
from bs4 import BeautifulSoup
import pandas as pd
from datetime import datetime as dt


headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3"
}
url = "https://en.wikipedia.org/wiki/List_of_S%26P_500_companies"

# send request to url
response = requests.get(url, headers=headers)
response.raise_for_status()

# get table
soup = BeautifulSoup(response.content, "html.parser")
table = soup.find("table", id="constituents")

df = pd.read_html(str(table))[0]

# Extract year added from Date added
df["Date added"] = pd.to_datetime(df["Date added"])
df["year_added"] = df["Date added"].dt.year

# group by year added and count companies added in that year
df_yearly_add = df[["year_added", "Symbol"]].groupby(
    'year_added'
  ).count().reset_index()

# filter to 2020 and above only
df_filtered = df_yearly_add[df_yearly_add.year_added >= 2020]
display(df_filtered)
print(df_filtered[df_filtered.Symbol == max(df_filtered.Symbol)])

/tmp/ipykernel_8414/1060976406.py:20: FutureWarning: Passing literal html to 'read_html' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  df = pd.read_html(str(table))[0]


,year_added,Symbol
51,2020,10
52,2021,10
53,2022,15
54,2023,15
55,2024,16
56,2025,18
57,2026,13


    year_added  Symbol
56        2025      18


Answer: From 2020 onwards, The year with the most companies added is 2025 with 18 additions.

### Additional: How many current S&P 500 stocks have been in the index for more than 20 years?

In [84]:
df_old_companies = df[df.year_added < dt.today().year - 20]
display(df_old_companies.head())
display(df_old_companies.shape[0])

### double check:
# ((dt.now().year - df.year_added) > 20).sum()

,Symbol,Security,GICS Sector,GICS Sub-Industry,Headquarters Location,Date added,CIK,Founded,year_added
0,MMM,3M,Industrials,Industrial Conglomerates,"Saint Paul, Minnesota",1957-03-04,66740,1902,1957
2,ABT,Abbott Laboratories,Health Care,Health Care Equipment,"North Chicago, Illinois",1957-03-04,1800,1888,1957
5,ADBE,Adobe Inc.,Information Technology,Application Software,"San Jose, California",1997-05-05,796343,1982,1997
7,AES,AES Corporation,Utilities,Independent Power Producers & Energy Traders,"Arlington, Virginia",1998-10-02,874761,1981,1998
8,AFL,Aflac,Financials,Life & Health Insurance,"Columbus, Georgia",1999-05-28,4977,1955,1999


218

Answer: There are 218 companies in the S&P500 that has been there for at least 20 years.

## Question 2. [Macro] Indexes YTD (as of 21 August 2026)

### How many indexes (out of 10) have better year-to-date returns than the US (S&P 500) as of August 21, 2026?